### ISSUE TITLE
How to forward declare sol::table / sol::state?

### ISSUE THREAD

**Issue by reworks** [ThePhD/sol2#544]
I've been using sol2 for a while now and I want to forward declare sol::table / sol::state, to avoid having to include it in header files, as the single header is quite large.
The problem is there seems to be a lot of template stuff and its hard to follow back to what in order to forward declare.

**Comment by ThePhD** [comment]
Look at `{root}/sol/forward.hpp`.

**Comment by ThePhD** [comment]
For the curious, that's this file: https://github.com/ThePhD/sol2/blob/develop/sol/forward.hpp

It includes `feature_test.hpp`. Note that `feature_test.hpp` includes nothing, so you can conceivably blend these two files together and have your own `sol_forward.hpp` that you include without including the whole single header file.

**Comment by ThePhD** [comment]
I have formalized this into the `single.py` generation header, and sol2's future releases will include a `sol_forward.hpp` header by default to make people's lives easier. Remember to put the same defines before the forward inclusion header as you would with the regular inclusion header.

https://github.com/ThePhD/sol2/tree/develop/single/sol

**Comment by reworks** [comment]
Thanks!

**Comment by reworks** [comment]
@ThePhD I'm getting a compilation error with the new sol_forward.hpp.

E.g: 
Source.hpp
Source.cpp

sol_forward.hpp is included in header, then I include sol2.hpp in source.cpp.

And I get the error  C2374	'sol::detail::default_safe_function_calls': redefinition; multiple initialization

**Comment by ThePhD** [comment]
@reworks Try again, it should work just fine.

**Comment by reworks** [comment]
Thanks!
